### DISCUSSION TITLE
Be serious about "open sourcing" Supabase.

### DISCUSSION THREAD

**Discussion by izzoa** [supabase/supabase#18109] [General]
Don't know where to start here...

Docker instructions do not work as written.

Surprising (if easily debuggable) errors in docker-compose.yml files.

No feature parity with hosted Supabase.

Open source isn't some marketing label you can just slap onto something, you actually have to make an actual effort...

**After observing myriad posted issues (unresolved, and worse yet - uncommented upon) - I'm not the only one frustrated and experiencing issues with your policies here.**

If your ultimate intent to "open source" Supabase is just a gateway to get people to pay for your hosted service, you should reconsider what it means to be "open source" at all.

**Comment by jocubeit** [comment]
I agree. My infatuation is waning.

Self-hosted doesn't seem to be a reality when we have to pull a repo and customize it. Ultimately if you're going to use Docker there should be a public image we can pull from a reputable registry. Docker compose is just not acceptable if we want to set up a Kubernetes cluster at the edge or on onsite. The local dev docker doesn't have parity with hosted Supabase, e.g. pgvector on hosted is at version 0.5 and local dev docker is at version 0.4 (hence no HNSW indexes on local dev). I've brought this up in discussions but I'm not getting any responses.

At this point I have to conclude that I either resign myself to the fact that I use hosted Supabase, or not at all.

**Comment by nicitaacom** [comment]
I not agree with @amizzo87
But I kinda agree with @jocubeit
About
> At this point I have to conclude that I either resign myself to the fact that I use hosted Supabase, or not at all.

You may don't use it - also I had an issue supabaseServer.ts and still no answer from voluteer who may help me but it doesn't meen that it should be closed source - anyway its supabase decide and its theirs privacy policy - create your own supabase and try to maintain all this stuff and only than write comments like this

**Comment by saltcod** [comment]
hey all


Thanks for the feedback, it's well-noted.  
We take our open source commitment seriously and I've raised this internally. 

I am going to move this Issue over to a Discussion so that we can follow up. If there are specific issues you'd like to raise, please feel free to put them in here as individual Issues and we'll triage them to the appropriate repos.

Our team is still very small, but we're committed to addressing feedback and maintaining feature parity between hosted and self-hosted versions. While there may be delays as we address feedback across 100+ repos, we aim for full feature parity where possible.

Regarding the Docker issue, we've reproduced it and created a tracking issue [here](https://github.com/supabase/supabase/issues/17630). We'll follow up there.

Let us know what else we can help with here, and please contribute back if you find anything that can be improved.

**Reply by izzoa** [reply]
Thank you, @saltcod . I appreciate the professional response and I'm happy to propose my own fixes and subsequent pull requests to help along the way, as well.

**Comment by kiwicopple** [comment]
hey all, sorry for the delay. i've been away from my keyboard for a week or so - just catching up on things.

First, I get it - there is room for improvement across every part of the supabase offering: from the self-hosted to the local development stack to the cloud offering. We could be more responsive in Discussions, or on twitter/reddit/discord/hn/emails/tickets. What's going unnoticed are the thousands of issues, discussions, support tickets, and pieces of feedback that we're responding to on a monthly basis. Yes, there will always be an issue/incident/PR which hasn't received enough attention - but I can tell you with 100% certainty that the team are working their hardest to support a burgeoning community. The reality is that the better we make the platform the faster the problems appear.

On this particular issue: 

> Docker instructions do not work as written. 

one of our devs just tried the local development instructions from top-to-bottom and there were no issues getting the self-hosted running. 

It's fair to bring us _actionable_ items which we can fix, with replication steps and error logs. It's even more digestible if you make an effort to contribute a fix yourself. The main benefit of open source is that it's a community effort - the company who develops the tools don't need to be a blocker when you have 100% control of the code (which is the case with everything we have developed).

On this point:

> Open source isn't some marketing label you can just slap onto something, you actually have to make an actual effort...

It's _not_ fair to accuse us of not being serious about open source. We are [fixing](https://github.com/supabase/supabase/pull/16097) the issues, providing free support, developing the products, running security audits, [sponsoring](https://opencollective.com/supabase) community efforts, supporting/developing existing open source tools like Postgres, PostgREST, Deno, and numerous postgres extensions.

If _we_ are not open source enough for you, I worry for the future of open source and the burden of expectations that maintainers need to meet to satisfy their community.

To @jocubeit - thank you for your contribution (to Wrappers). I've responded to your comment [here](https://github.com/orgs/supabase/discussions/17930#discussioncomment-7268200). I'll follow up with some docs updates shortly. If there are any other features you're missing it's probably just a version bump, or a feature flag that we need to turn off (which you can also see in the code). Feel free to open an issue/discussion or ping me directly and I'll jump into it when I'm back at my desk.

**Reply by jocubeit** [reply]
@kiwicopple thank you for your professional responses. Supabase is an awesome product overall, and I enjoy using it. I'll check out your comments.

**Reply by vickkhera** [reply]
I've been in the open source community since we just called it free software in the late 80's. We used to download the source as shell archive files (shar) from usenet feeds. I've presented many times at O'Reilly OSCON. Never have I ever heard such a whiny complaint that someone doesn't provide a working binary docker container for their software to make it easy to use. That's not what Open source is about. It is about the community and developing in the open and allowing people to use your source code. It is not about doing the work to integrate the software into your use case. Supabase clearly meets the intentions of open source, unlike some large companies (looking at you, Apple) who drop code into the web then hide all their development process and bug trackers behind their corporate walls.

**Reply by jocubeit** [reply]
@vickkhera I hardly consider it a whiny complaint that Supabase purports to provide a self-hosted option but only provides a dev solution. Docker Compose is distributed with Docker Desktop and is intended for development environments - it is not a solution for production environments. Creating our own images means everyone has to build their own version, which could lead to more issues being reported to Supabase staff. A company-issued official docker image would reduce any disparity, leading to a single-supported version. A simple case in point is my comment that the docker compose version didn't have parity with hosted Supabase - how can we develop in dev and push to hosted when the components are different?

You and I come from the same era, and I've been a massive supporter and contributor to OSS, which is demonstrated in my GitHub, StackOverflow and LinkedIn profiles. Open Source is rarely a philosophical endeavour any more, it's a business strategy. Downloading source from usenet feeds and presenting at OSCON brings nothing to this discussion, other than to attempt to elevate yourself in the eyes of readers.

**Comment by izzoa** [comment]
Apologies myself now, been away from the keyboard for a bit.

Thanks, all, for your profesional replies. My point with this comment was **not** to bring a laundry list of bugs - can easily do that with pull requests and diagnosable bug reports through proper avenues - but rather, this was meant to spark more of a **high-level discussion** on the following:

- Critique over how the dichotomy between how paid Supabase and open-source Supabase is handled and presented, from a marketing and documentation perspective.
- _How_ open source-committed Supabase is (vs. companies like Apple who _do_ just use it as a "marketing label").

Respectfully, please understand that as the intent, as treat it as such.

I'd like to address a few salient comments.

@kiwicopple 
>one of our devs just tried the local development instructions from top-to-bottom and there were no issues getting the self-hosted running.

When I wrote that initial comment, it was correct that the instructions did _not_ work as written because of [this](https://github.com/supabase/supabase/commit/599bf0ea764acd0a4c5007641ae814fff3e1f635) typo:
<img width="1414" alt="image" src="https://github.com/supabase/supabase/assets/5224781/355bc056-82cd-4271-85d8-f961432be3e0">

It has since been fixed (thank you). That isn't the only bug (logstash has a few more, for which I'm preparing a pull request for) but also, as mentioned above, this isn't the forum I am attempting to advance bug reports. 

> If we are not open source enough for you, I worry for the future of open source and the burden of expectations that maintainers need to meet to satisfy their community.

While I fully respect the effort Supabase has put in - and that of any company - any effort put in to releasing / open sourcing their product, what I do not respect are expectations that any contribution "has to be a pull request" or "fully diagnosable bug report." You can even see some evidence of that within this thread. 

I'm attemping to spark a discussion. My intent is **not** to have some easy `chmod +x && ./magic-run-everything.sh` script that I can just "download-and-go"; I'm happy to put in the effort on my own, I just want a firm base on which to put in that work.

Now, by my "standards," you could commit once a decade and it could be a function that echos `1+1=2`. I wouldn't care. Fine, cool, awesome. What I _would_ expect is that when a company provides a "self-hosted" version of their product, it is feature-comparable _unless otherwise stated_. Please show me where a reasonable person can assume the hosted and self-hosted are _not_ feature comparable based on the [documentation](https://supabase.com/docs/guides/self-hosting):

<img width="380" alt="image" src="https://github.com/supabase/supabase/assets/5224781/cd98e1f6-2a40-4ea1-a43c-9ffc275e3bb9">

- I would suggest making a table or something similar that compares the capabilities (intended and/or implemented) of self-hosted Supabase with paid Supabase, so that a reasonable person can see they are not currently feature equal or at least relatively so. A lot of my frustration with "self-hosted" was the lack of parity and that is not mentioned in the docs (unless I missed it, which is possible...).

Again, this is all meant to spark a debate and reasonable discussion - not meant as a personal attack - so please do not take it as such. I love that Supabase has started putting in the effort to open source and if I did not care about the product and if I did not want to see it improve and grow, I would not be spending so much time writing this critique.

---

Appendix:

@vickkhera 
> I've been in the open source community since we just called it free software in the late 80's. We used to download the source as shell archive files (shar) from usenet feeds. I've presented many times at O'Reilly OSCON. Never have I ever heard such a whiny complaint that someone doesn't provide a working binary docker container for their software to make it easy to use. That's not what Open source is about. It is about the community and developing in the open and allowing people to use your source code. It is not about doing the work to integrate the software into your use case. 

Listen, @vickkhera, everything evolves over time - especially in a field as fast-moving as software development. If have been in the open source community since the 80's, you should know that. As a meta-point, "open source" clearly no longer means what it did (to which you alluded), and that is fine. My complaint was that their instructions did not work as written. If `man` pages of something were inaccurate, would you not also?

> Supabase clearly meets the intentions of open source, unlike some large companies (looking at you, Apple) who drop code into the web then hide all their development process and bug trackers behind their corporate walls.

While Apple does have [some](https://opensource.apple.com/) open source projects, the heart of what Apple does is closed source, and the software (MacOS, iOS) powering their core products (Mac, iPhone) is absolutely closed source.

**Reply by saltcod** [reply]
Thanks @amizzo87 

Want to clarify this one piece: 
> what I do not respect are expectations that any contribution "has to be a pull request" or "fully diagnosable bug report." 

We definitely don't expect this from our community members. Quite the opposite is true actually. We actively solicit feedback in its roughest form from all channels: Twitter, Discord, Github, Threads, etc. Basically anywhere we can get feedback, we take it. Our main [main repo](https://github.com/supabase/supabase/issues) has many examples of us triaging, investigating and fixing vague, fuzzy and hard to reproduce bug reports. 

We don't expect that all contributions be pull requests, nor that they be fully diagnosable bug reports. Ideally, a fully diagnosable bug report will come with a community pull request, but that's definitely not the expectation. We have an incredible community of contributors and the last thing we want is to put roadblocks in front of contributions. Of course easily reproducible bugs are the easiest ones to fix, but it's not a hard rule for any of our repos. 

Appreciate the feedback.
